egs/wsj/s5/utils/data/internal/combine_segments_to_recording.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
```

Comment:
For new python scripts we prefer python3.   (and you can remove compatibility things).  This will mean it's not necessary to verify separately that it's python3 compatible.

Reply:
As discussed, we'll remove the text-manipulating aspects of this script and replace them with a separate call to apply_map.pl.
Unfortunately the way the text formats for Kaldi are designed don't mesh well with python3.  The "text" file format is encoding-agnostic: any encoding is allowed, as long as (when interpreted as a bytestream) whitespace characters represent space, and newline represents newline.  And because it's never needed by Kaldi itself, we never have to specify the encoding. 
It might actually be possible to deal with this in python3 using its 'byte-string' type.  From a search it looks like it is possible to do things like splitting byte-strings.  This might be a good test case for figuring out the right way to do this in python, because as python3 becomes more and more the default, we need to make sure it doesn't break things.